internal/vault/credentials/helpers.go

```diff
@@ -47,3 +47,15 @@ func getSecret(ctx context.Context, client ctrlclient.Client, key ctrlclient.Obj
 
 	return secret, nil
 }
+
+func getSA(ctx context.Context, client ctrlclient.Client, saKey ctrlclient.ObjectKey) (*corev1.ServiceAccount, error) {
```

Comment:
nit: can we rename `saKey` to `key` like in `getSecret()` or  `objKey` is also an option for both functions.